src/librustc_typeck/check/generator_interior.rs

```diff
@@ -54,12 +54,15 @@ impl<'a, 'gcx, 'tcx> InteriorVisitor<'a, 'gcx, 'tcx> {
             debug!("type in expr = {:?}, scope = {:?}, type = {:?}, count = {}, yield_span = {:?}",
                    expr, scope, ty, self.expr_count, yield_span);
 
-            if self.fcx.any_unresolved_type_vars(&ty) {
-                let mut err = struct_span_err!(self.fcx.tcx.sess, source_span, E0698,
-                    "type inside generator must be known in this context");
-                err.span_note(yield_span,
-                              "the type is part of the generator because of this `yield`");
-                err.emit();
+            if let Some((unresolved_type_span, unresolved_type)) =
+                self.fcx.get_unresolved_type_vars(&ty).pop()
```

Comment:
It doesn't really matter, but it seems like we are keeping a whole vector but only using the last entry -- maybe we should just keep an option? Or, meh, keep the vector, maybe we'll use the data later. This only executes in the case of error anyway.

Reply:
The previous finder commented that "They get pushed into a vector.", though there was no vector on that struct. I couldn't spot a point when it did have a vector, but I guess someone previously optimized the vector away and just left the bool (i.e. the existence of an unbound type). Given this, maybe we should just return details of the first unbound type.

Reply:
We should definitely not keep the vec as the visitor returns true at that point halting the visit so we would never get more than one in any case.